client/src/Attendance/AttendanceDataCell.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { Checkbox, TimePicker } from 'antd'
+import { useTranslation } from 'react-i18next'
+
+export default function AttendanceDataCell({
+  record,
+  columnIndex,
+  updateAttendanceData
+}) {
+  const { t } = useTranslation()
+  const [absence, setAbsence] = useState(null)
+  const [checkInSelected, setCheckInSelected] = useState(false)
+  const [checkOutSelected, setCheckOutSelected] = useState(false)
+  const handleChange = (update = {}, cb = () => {}) => {
+    updateAttendanceData(update, record, columnIndex)
+    cb()
+  }
+
+  return (
+    <div className="flex m-">
+      <div className="mr-4">
+        <p className="font-proxima-nova-alt font-semibold mb-1">
+          {t('checkIn').toUpperCase()}
+        </p>
+        <TimePicker
+          className="w-20 h-8"
+          use12Hours={true}
+          format="h:mm a"
+          disabled={absence}
+          onChange={(m, ds) =>
+            m
+              ? handleChange({ check_in: ds }, setCheckInSelected(true))
+              : handleChange({ check_in: '' }, setCheckInSelected(false))
+          }
+          suffixIcon={null}
+        />
+      </div>
+      <div className="mr-4">
+        <p className="font-proxima-nova-alt font-semibold mb-1">
+          {t('checkOut').toUpperCase()}
+        </p>
+        <TimePicker
+          className="w-20 h-8"
+          use12Hours={true}
+          format="h:mm a"
+          disabled={absence}
+          onChange={(m, ds) =>
+            m
+              ? handleChange({ check_out: ds }, setCheckOutSelected(true))
```

Comment:
What do `m` and `ds` stand for here?

Reply:
m is the moment dateobject (moment bc that's what antdesign uses). ds is the datestring of the selected date. renaming to be clear.